libs/api/domains/auth/src/lib/services-v2/meDelegations.service.ts

```diff
@@ -0,0 +1,170 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import differenceWith from 'lodash/differenceWith'
+
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import {
+  MeDelegationsApi,
+  MeDelegationsControllerFindAllDirectionEnum,
+  MeDelegationsControllerFindAllValidityEnum,
+} from '@island.is/clients/auth/delegation-api'
+
+import {
+  CreateDelegationInput,
+  DelegationInput,
+  DelegationsInput,
+  DeleteDelegationInput,
+  PatchDelegationInput,
+  UpdateDelegationInput,
+} from '../dto'
+import { DelegationByOtherUserInput } from '../dto/delegationByOtherUser.input'
+import { DelegationDTO, MeDelegationsServiceInterface } from '../services/types'
+
+@Injectable()
+export class MeDelegationsServiceV2 implements MeDelegationsServiceInterface {
+  constructor(private delegationsApi: MeDelegationsApi) {}
+
+  private delegationsApiWithAuth(auth: Auth) {
+    return this.delegationsApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  getDelegations(
+    user: User,
+    input: DelegationsInput,
+  ): Promise<DelegationDTO[]> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerFindAll({
+      domain: input.domain,
+      direction: MeDelegationsControllerFindAllDirectionEnum.Outgoing,
+      validity: MeDelegationsControllerFindAllValidityEnum.IncludeFuture,
+    })
+  }
+
+  getDelegationById(
+    user: User,
+    { delegationId }: DelegationInput,
+  ): Promise<DelegationDTO | null> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerFindOne({
+      delegationId,
+    })
+  }
+
+  async getDelegationByOtherUser(
+    user: User,
+    { toNationalId, domain }: DelegationByOtherUserInput,
+  ): Promise<DelegationDTO | null> {
+    const delegations = await this.delegationsApiWithAuth(
+      user,
+    ).meDelegationsControllerFindAll({
+      direction: MeDelegationsControllerFindAllDirectionEnum.Outgoing,
+      xQUERYOTHERUSER: toNationalId,
```

Comment:
This is generated like this from OpenAPI "X-QUERY-OTHER-USER". Though I guess it's worth trying to register the header with different casing (eg "X-Query-Other-User"). It shouldn't matter when accessing it since that's case insensitive.